pkg/backend/apply.go

```diff
@@ -166,14 +166,15 @@ func PreviewThenPrompt(ctx context.Context, kind apitype.UpdateKind, stack Stack
 	}
 
 	// Otherwise, ensure the user wants to proceed.
-	plan, err = confirmBeforeUpdating(ctx, kind, stack, op, events, plan, explainer)
+	plan, err = confirmBeforeUpdating(ctx, kind, stack.Ref(), op, events, plan, explainer)
 	close(eventsChannel)
 	return plan, changes, err
 }
 
 // confirmBeforeUpdating asks the user whether to proceed. A nil error means yes.
-func confirmBeforeUpdating(ctx context.Context, kind apitype.UpdateKind, stack Stack, op UpdateOperation,
+func confirmBeforeUpdating(ctx context.Context, kind apitype.UpdateKind, stackRef StackReference, op UpdateOperation,
 	events []engine.Event, plan *deploy.Plan, explainer Explainer,
+	askOpts ...survey.AskOpt,
```

Comment:
Are `askOpts` purely for the test? Why does the test require them?

Reply:
Yep! So we can pipe control stdin/out and assert on them. Will add a comment.

Easiest way to verify this change manually (and could remove the test if too intrusive), is to reduce the timeout down to 1s

https://github.com/pulumi/pulumi/blob/d240ac72ba95ae1faec2122a5787ec69f108cf45/pkg/backend/httpstate/client/client.go#L1440-L1441

Reply:
Added a comment here, also tweaked timeout error message and added some color.